### PR TITLE
work with file

### PR DESCRIPTION
**Pull request by Danylo056**
Added class TestFunctions with new method for creating temp file

### CODE REVIEW

File: functions/test_functions.rb [hunk 1]
```diff
@@ -1,25 +1,46 @@
...
+  #
+  #
+  def create_temp_file(path, name)
+    dir = File.dirname(path)
+
+    unless File.directory?(dir)
+      FileUtils.mkdir_p(dir)
+    end
+
+    path << '#{name}'
```
Comment (dimahadgi) [inline functions/test_functions.rb]: Use File.join here

File: functions/test_functions.rb [hunk 2]
```diff
@@ -1,25 +1,46 @@
...
+class TestFunctions
+  def initialize(path, name)
+    @path = path
+    @name = name
+    create_file(@path, @name)
   end
+  ##
+  # Create temp file in selected directory
+  #
+  # @param path [String] path to temp file
```
Comment (dimahadgi) [inline functions/test_functions.rb]: add name argument

File: functions/test_functions.rb [hunk 3]
```diff
@@ -1,25 +1,46 @@
...
-end
-##
-# Delete directory including all files
-#
-# @param path [String] path to temp directory
-#
-#
-def delete_temp_file(path)
-  FileUtils.rm_rf(path)
+    a = File.new(path, 'w+')
```
Comment (dimahadgi) [inline functions/test_functions.rb]: use FileUtils

File: functions/test_functions.rb [hunk 4]
```diff
@@ -1,25 +1,46 @@
...
+    @name = name
+    create_file(@path, @name)
   end
+  ##
+  # Create temp file in selected directory
+  #
+  # @param path [String] path to temp file
+  #
+  #
+  def create_temp_file(path, name)
```
Comment (dimahadgi) [inline functions/test_functions.rb]: This function should be privat

Comment (dimahadgi) [reply functions/test_functions.rb]: if we call it only in the scope of this class

File: functions/test_functions.rb [hunk 5]
```diff
@@ -1,25 +1,46 @@
...
+    a = File.new(path, 'w+')
+    a.close
+  end
+  ##
+  # Delete directory including all files
+  #
+  # @param path [String] path to temp directory
+  #
+  #
+  public
```
Comment (dimahadgi) [inline functions/test_functions.rb]: this probably useless, you can call methods without this

File: functions/test_functions.rb [hunk 6]
```diff
@@ -1,25 +1,46 @@
...
+    FileUtils.rm_rf(path)
+  end
+  ##
+  # Create temp file with selected name in selected directory
+  #
+  # @param path [String] path to temp file
+  # @param name [String] name for temp file
+  #
+  #
+  def create_file(path, name)
```
Comment (dimahadgi) [inline functions/test_functions.rb:24]: remove first function if this is the main one

File: spec/test_spec.rb [hunk 7]
```diff
@@ -1,90 +1,90 @@
...
   it 'SPAWN and INCLUDE' do
-    expected_data = 'Approximate round trip times in milli-seconds:'
-    spn = spawn('ping localhost', :out=>['C:\test_folder\temp_file.txt', "w+"])
+    #expected_data = 'Approximate round trip times in milli-seconds:'
+    expected_data = TestData::EXPECTED_APPROXIMATE
+    spn = spawn(TestData::PING_LOCALHOST, :out=>[TestData::TEMP_FILE_PATH, TestData::OUTWRITEPLUS])
     Process.wait spn
     #success?
-    file = File.open('C:\test_folder\temp_file.txt')
+    file = File.open(TestData::TEMP_FILE_PATH)
```
Comment (dimahadgi) [inline spec/test_spec.rb]: replace with FileUtils

File: spec/test_spec.rb [hunk 8]
```diff
@@ -1,90 +1,90 @@
...
 
   it 'SPAWN and END WITH' do
-    expected_data = 'Average = 0ms'
-    spn = spawn('ping localhost', :out=>['C:\test_folder\temp_file.txt', "w+"])
+    expected_data = TestData::EXPECTED_AVERAGE
+    spn = spawn(TestData::PING_LOCALHOST, :out=>[TestData::TEMP_FILE_PATH, TestData::OUTWRITEPLUS])
     Process.wait spn
     #success?
-    file = File.open('C:\test_folder\temp_file.txt')
+    file = File.open(TestData::TEMP_FILE_PATH)
```
Comment (dimahadgi) [inline spec/test_spec.rb]: Same here

File: spec/test_spec.rb [hunk 9]
```diff
@@ -1,90 +1,90 @@
...
     file_data = file.read.strip
     expect(file_data).to end_with(expected_data)
     file.close
   end
 
 
   it 'SPAWN and START WITH' do
-    expected_data = 'Pinging DESKTOP-MRO0O4C'
-    spn = spawn('ping localhost', :out=>['C:\test_folder\temp_file.txt', "w+"])
+    #expected_data = 'Pinging DESKTOP-MRO0O4C'
```
Comment (dimahadgi) [inline spec/test_spec.rb]: do not forget to clean this up

File: spec/test_spec.rb [hunk 10]
```diff
@@ -1,90 +1,90 @@
...
+    expected_data = TestData::EXPECTED_DESKTOP
+    spn = spawn(TestData::PING_LOCALHOST, :out=>[TestData::TEMP_FILE_PATH, TestData::OUTWRITEPLUS])
     Process.wait spn
     #success?
-    file = File.open('C:\test_folder\temp_file.txt')
+    file = File.open(TestData::TEMP_FILE_PATH)
     file_data = file.read.strip
```
Comment (dimahadgi) [inline spec/test_spec.rb]: same here

File: test_data/test_data.rb [hunk 11]
```diff
@@ -0,0 +1,11 @@
+module TestData
+  EXPECTED_APPROXIMATE = 'Approximate round trip times in milli-seconds:'.freeze
+  EXPECTED_AVERAGE = 'Average = 0ms'
+  EXPECTED_DESKTOP = 'Pinging DESKTOP-MRO0O4C'
+  EXPECTED_STATISTICS = 'Ping statistics for ::1:'
+  EXPECTED_HELLO = 'Hello World'
+  TEMP_FILE_PATH = "C:\\test_folder\\temp_file.txt"
+  PING_LOCALHOST = 'ping localhost'
+  REGTEMPLATE = /Minimum = \d+ms, Maximum = \d+ms, Average = \d+ms/
+  OUTWRITEPLUS = 'w+'
```
Comment (dimahadgi) [inline test_data/test_data.rb]: this is to match :) move it back

Comment (dimahadgi) [inline test_data/test_data.rb]: for grouped data we can use hashes
eg 
       
EXPECTED_DATA = {
     DESKTOP = 'Pinging DESKTOP-MRO0O4C',
     HELLO = 'Hello World'
}

**Review by dimahadgi — COMMENTED** [review]
Please add rpsec run to the description of the PR(screenshot should be good enough)

File: functions/test_functions.rb (hunk 1 above)
Comment (Danylo056) [reply functions/test_functions.rb]: Done.

File: functions/test_functions.rb (hunk 3 above)
Comment (Danylo056) [reply functions/test_functions.rb]: Done.

File: functions/test_functions.rb (hunk 2 above)
Comment (Danylo056) [reply functions/test_functions.rb]: Done.

File: functions/test_functions.rb (hunk 4 above)
Comment (Danylo056) [reply functions/test_functions.rb]: Done.

File: spec/test_spec.rb (hunk 8 above)
Comment (Danylo056) [reply spec/test_spec.rb]: Done.

File: functions/test_functions.rb (hunk 5 above)
Comment (Danylo056) [reply functions/test_functions.rb]: Done.

File: spec/test_spec.rb (hunk 9 above)
Comment (Danylo056) [reply spec/test_spec.rb]: Done.

File: test_data/test_data.rb (hunk 11 above)
Comment (Danylo056) [reply test_data/test_data.rb]: Done.

Comment (Danylo056) [reply test_data/test_data.rb]: Done.

File: spec/test_spec.rb (hunk 10 above)
Comment (Danylo056) [reply spec/test_spec.rb]: Done.

File: functions/test_functions.rb (hunk 6 above)
Comment (Danylo056) [reply functions/test_functions.rb:24]: Done.

File: spec/test_spec.rb (hunk 7 above)
Comment (Danylo056) [reply spec/test_spec.rb]: Done.

File: test_data/test_data.rb [hunk 12]
```diff
@@ -1,11 +1,10 @@
...
   TEMP_FILE_PATH = "C:\\test_folder\\temp_file.txt"
-  PING_LOCALHOST = 'ping localhost'
-  REGTEMPLATE = /Minimum = \d+ms, Maximum = \d+ms, Average = \d+ms/
-  OUTWRITEPLUS = 'w+'
+  EXPECTED_DATA = {
+    :EXPECTED_APPROXIMATE => 'Approximate round trip times in milli-seconds:',
+    :EXPECTED_AVERAGE => 'Average = 0ms',
+    :EXPECTED_DESKTOP => 'Pinging DESKTOP-MRO0O4C',
+    :EXPECTED_STATISTICS => 'Ping statistics for ::1:',
+    :EXPECTED_HELLO => 'Hello World',
```
Comment (dimahadgi) [inline test_data/test_data.rb]: in hash if you use symbols - don't use all caps
for instance
HASH = {
    :key => 'value'
}

File: spec/test_spec.rb [hunk 13]
```diff
@@ -33,58 +30,51 @@
...
     file.close
   end
 
   it 'REGEX' do
-    spn = spawn(TestData::PING_LOCALHOST, :out=>[TestData::TEMP_FILE_PATH, TestData::OUTWRITEPLUS])
+    spn = spawn('ping localhost', :out=>[TestData::TEMP_FILE_PATH, 'w+'])
     Process.wait spn
     #success?
     file = File.open(TestData::TEMP_FILE_PATH)
     file_data = file.read.strip
```
Comment (dimahadgi) [inline spec/test_spec.rb]: use File.Utils across all test spec

**Review by dimahadgi — APPROVED** [review]
Resolve remaining comments

File: test_data/test_data.rb (hunk 12 above)
Comment (Danylo056) [reply test_data/test_data.rb]: Done.

File: spec/test_spec.rb (hunk 13 above)
Comment (Danylo056) [reply spec/test_spec.rb]: Done.

**Pull request merged**